test/PrintAsObjC/availability.swift

```diff
@@ -45,11 +45,29 @@
 // CHECK-DAG: SWIFT_AVAILABILITY(ios_app_extension,unavailable)
 // CHECK-DAG: SWIFT_AVAILABILITY(tvos_app_extension,unavailable)
 // CHECK-DAG: SWIFT_AVAILABILITY(watchos_app_extension,unavailable)
+// CHECK-DAG: ;
+// CHECK-NEXT: - (void)deprecatedMethodRenamedToOverloadMethod1WithFirst:(NSInteger)first second:(NSInteger)second SWIFT_DEPRECATED_MSG("", "overloadMethodWithFirst:second:");
+// CHECK-NEXT: - (void)overloadMethodWithFirst:(NSInteger)first second:(NSInteger)second;
+// CHECK-NEXT: - (void)overloadMethod2WithFirst:(double)first second:(double)second;
+// CHECK-NEXT: + (void)deprecatedAvailabilityWithValue:(NSInteger)value;
+// CHECK-NEXT: - (void)deprecatedInstanceMethodRenamedToClassMethodWithValue:(NSInteger)value SWIFT_DEPRECATED_MSG("This method has a renamed attribute point to class method instead of a instance method. It should show the Swift name here", "classMethodWithACustomObjCName(x:)");
+// CHECK-NEXT: + (void)deprecatedClassMethodRenamedToInstanceMethodWithValue:(NSInteger)value SWIFT_DEPRECATED_MSG("This method has a renamed attribute point to instance method instead of a class method. It should show the Swift name here", "instanceMethodWithACustomObjCName(x:)");
+// CHECK-NEXT: - (void)customObjCNameInstanceMethodWithX:(NSInteger)x;
+// CHECK-NEXT: + (void)customObjCNameClassMethodWithX:(NSInteger)x;
+// CHECK-NEXT: + (void)makeDeprecatedAvailabilityWithValue:(NSInteger)value SWIFT_DEPRECATED_MSG("use something else", "deprecatedAvailabilityWithValue:");
+// CHECK-NEXT: + (void)unavailableAvailabilityWithValue:(NSInteger)value;
+// CHECK-NEXT: + (void)makeUnavailableAvailabilityWithValue:(NSInteger)value
+// CHECK-DAG: SWIFT_UNAVAILABLE_MSG("'__makeUnavailableAvailability' has been renamed to 'unavailableAvailabilityWithValue:': use something else");
```

Comment:
CHECK-SAME would be a better choice here too (since in this case there's only one thing to match).